app/src/main/res/values/strings.xml

```diff
@@ -190,6 +190,8 @@
   <string name="read_storage_permission_rationale">Required permission: Read external storage. App cannot access your gallery without this.</string>
   <string name="write_storage_permission_rationale">Required permission: Write external storage. App cannot access your camera/gallery without this.</string>
   <string name="location_permission_title">Requesting Location Permission</string>
+  <string name="in_app_camera_location_permission_title">Record location for in-app shots</string>
+  <string name="in_app_camera_location_switch_pref_summary">Enable this to record location while using in-app camera</string>
```

Comment:
I think we're explaining this in the in-app camera pop-up already.

Reply:
The user has the liberty to toggle the setting anytime they want. So, it's better to be unambiguous in the setting description.

It's better not to presume that the user would remember (or) fully grasp what they saw in a one-time dialog.

Reply:
I added this as a follow-up for the dialog (for privacy reasons) as discussed above. As a privacy setting, this summary sounded enough to me but in case we're adding this as an in-app camera related setting, should I move it to the General section instead of the Privacy section?

Reply:
> As a privacy setting, this summary sounded enough to me

If this setting were really controlling whether the app records location for in-app shots, the description is totally fine. The problem is that this setting only controls whether the app appends the location when the in-built camera doesn't provide us an location.

> ... in case we're adding this as an in-app camera related setting ...

Isn't this a in-app camera related setting already? What am I missing ? 🤔 

> should I move it to the General section instead of the Privacy section?

Moving it out of "Privacy" is a nice idea. Moving it into "General" isn't. That section is already too cluttered. I think we could introduce a seprarate section title named "Uploads" and move the following preferences into that section:

- Save In-app shots
- Use custom author name
- Custom author name
- Photo picker one
- Record location with in-app shots (with a tweaked description)

This might be a better situation help de-clutter the settings and position this setting in a better place 🙂

Reply:
> This could also occur in cases where the app has location access and location is turned on the device but saving location is turned off in the device camera. The user could've done this intentionally. The app obtained location would be uploaded in this case without the user having any idea about it.

I'd added it as a privacy setting for this reason. Adding a separate section sounds better. Thank you for this great suggestion!

> The problem is that this setting only controls whether the app appends the location when the in-built camera doesn't provide us an location.

I'm sorry if I made it too abstract, seems like I still need to understand a lot about user behaviour. Here's what I took into consideration: 

As a user, I wouldn't have thought if the camera is recording the location or the app is doing it. As a _new_ user, if I open the Settings first and like my pictures to be geotagged, I'll simply turn it on. And if I open the in-app camera first, I'll still turn it on to be on the safer side. I don't think naive users will test if the default camera app records location, for them, it's always our app that's not working properly (even if that's their camera's issue). In fact, the notion of default camera app exists only for us because we know we're triggering it using an intent. For users, it's just the app's camera.

Thank you for explaining it in detail, will add a separate section with modified summary 👍

Reply:
I'll add a new section for the in-app shots location. Created a new issue (#5258) for the rest of the options as these are not a part of this issue. Will handle the rest of them there, I hope it sounds okay.

Reply:
> I'll add a new section for the in-app shots location. Created a new issue (#5258) for the rest of the options as these are not a part of this issue. Will handle the rest of them there, I hope it sounds okay.

To be honest, having a separate section with just the setting for in-app shots location is what sounds awkward to me. While it might seem a bit unrelated, doesn't it seem too trivial to be handled in a separate issue ? You'll only be doing the change as a part of a related change 🤔 

The primary thing of concern is the section itself. A section for in-app shot location alone doesn't seem useful. The kind of categorization I suggest helps gather related settings and sounds reasonable. Or am I missing something ?

Reply:
I'm sorry if this was not intended😔

Reply:
You don't have to apologize for everything. All these are part of the contribution cycle. You just need to be sure to learn along the way 🙂 

In case it helps, give the following article a read: [How to Become A Hacker](http://web.archive.org/web/20230220105955/https://www.catb.org/~esr/faqs/hacker-howto.html#what_is)